docs/SocialLogin.md

```diff
@@ -0,0 +1,368 @@
+        else no token storage is implemented and zot only contains original token
+        Zot Server ->> Zot Server: decrypt Zot server generated token and extract underlying auth_token, obrain information from auth providers
+        Zot Server ->> Social Server : verifies token validity and origin by calling every social provider API (if available)
+        end
+        Note left of Social Server: this verification will also be done for clients like Skopeo, which use PAT instead of Zot tokens
+        alt token invalid
+            Zot Server ->> Client app: REQ - status code 401 Unauthorized
+        else token valid
+            Zot Server ->> Client app: give resources specific to user
+        end
+```
+
+### 4: Backend with session cookies
+
+#### Short overview of the workflow
+
+    - Similar to flow#3
+    - Instead of generating JWT tokens use sessions to store information
+
+#### zui needs the following features from the zot backend
+
+    - Endpoints needed are the same as flow #3.
+    - Needs session tracking on zot side.
+
+#### Sequence diagram for this workflow
+
+```mermaid
+    sequenceDiagram
+        actor User
+        Note right of User: representation of user interactions outside Client app
+        User ->> Client app: click "sign in with X" button
+        Client app->>Zot Server: http://zot.server.domain:port/auth/gh/login
+        participant Zot Server 
+        Note right of Zot Server: query parameters such as STATE, CLIENT_ID, SCOPE and CALLBACK_URL are added to https://social-provider.com/login/oauth/authorize
+        Zot Server->>Social Server:  https://social-provider.com/login/oauth/authorize
+        Social Server ->> User: user gets the social login screen (ex: github login) 
+        User ->> Social Server: user logs to the social server
+        Social Server ->> Zot Server: social server calls http://zot.server.domain:port/auth/gh/callback
+        Note left of Social Server: redirect to callback url, attaching state and code as query params(?code=1234567890&state=STATE)
+        Zot Server ->> Social Server: REQ - https://social-provider.com/login/oauth/token (gets token using code from prev request)
+        Social Server ->> Zot Server: RESP - returns access_token, refresh_token, period of validity using previously returned code as a query param
+        Zot Server -->> Zot Server : creates session token from access_token and updates sessions map
+        Note right of Zot Server: cookie containing user data extracted from token
+        Zot Server ->> Client app: RESP - /auth/gh/login - set session token as cookie
+        Client app->>Zot Server: send cookie containing session token, asking for resource
+        Zot Server -->> Zot Server : verifies session token exists in sessions map
+        alt session token  exists
+            alt access_token valid
+            Note right of Zot Server: we may or we may not implement this, as we already got the user data before generating the token
```

Comment:
Does this mean it doesn't matter if the access_token is expired? do we only care about the fact that the first access_token we get is enough to get user:email and other useful data ?

Reply:
Yes, that's what I was thinking.
As long as our session token is not expired we don't really care about expiry of access_token.
Assuming the session token is designed to expire in a reasonable timeframe.